UNITTESTS/drivers/Watchdog/test_watchdog.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "drivers/Watchdog.h"
+using ::testing::NiceMock;
+using ::testing::AtLeast;
+
+
+class mock_mbed_assert {
+public:
+    MOCK_METHOD0(process,void());
+};
+
+class TestWatchdog : public testing::Test{
+protected:
+    virtual void SetUp() {}
+    virtual void TearDown() {}
+};
+
+NiceMock<mock_mbed_assert> mwatchdog;
+
+void mbed_assert_internal(const char *expr, const char *file, int line) {
+    mwatchdog.process();
+}
+TEST_F(TestWatchdog, wdog_constructor)
+{
+    mbed::Watchdog *watchdog = new mbed::Watchdog(500,"watchdog_unittest");
+    EXPECT_TRUE(watchdog);
+    delete watchdog;
+}
+
+
+TEST_F(TestWatchdog, wdog_start_pass)
+{
+    mbed::Watchdog *watchdog = new mbed::Watchdog(500,"watchdog_unittest");
+    EXPECT_TRUE(watchdog);
+    EXPECT_CALL(mwatchdog, process()).Times(0);
+    watchdog->start();
+    watchdog->stop();
+    delete watchdog;
+}
+
+TEST_F(TestWatchdog, wdog_start_fail)
+{
+    mbed::Watchdog *watchdog = new mbed::Watchdog(500,"watchdog_unittest");
+    EXPECT_TRUE(watchdog);
+    EXPECT_CALL(mwatchdog, process()).Times(AtLeast(1));
+    watchdog->start();
+    watchdog->start();
+    watchdog->stop();
+    delete watchdog;
+}
+
```

Comment:
It may be good to add a test to confirm that the watchdog fire at the correct time if kick is missed.